ads/google/a4a/utils.js

```diff
@@ -115,9 +116,14 @@ export function googleAdUrl(
     const viewportRect = viewport.getRect();
     const iframeDepth = iframeNestingDepth(win);
     const viewportSize = viewport.getSize();
-    if (ValidAdContainerTypes.indexOf(adElement.parentElement.tagName) >= 0) {
-      queryParams.push({name: 'amp_ct',
-                        value: adElement.parentElement.tagName});
+    let parentElement = adElement.parentElement;
+    const containerTypes = [];
+    while (parentElement && ValidAdContainerTypes[parentElement.tagName]) {
```

Comment:
Are we confident tagName is always uppercase or should we consider parentElement.tagName.toUpperCase?